github/ci/prow/files/jobs/nmstate/kubernetes-nmsate-presubmits.yaml

```diff
@@ -0,0 +1,54 @@
+---
+presubmit_base: &presubmit_job_base
+  skip_branches:
+    - release-\d+\.\d+
+  annotations:
+    fork-per-release: "true"
+  always_run: true
+  optional: false
+  decorate: true
+  decoration_config:
+    timeout: 3h
+    grace_period: 5m
+  max_concurrency: 6
+  labels:
+    preset-dind-enabled: "true"
+    preset-docker-mirror: "true"
+    preset-shared-images: "true"
+  spec: &presubmit_spec_base
+    nodeSelector:
+      type: bare-metal-external
+    containers:
+      - &presubmit_runner_container_base
+        image: gcr.io/k8s-testimages/bootstrap:v20190516-c6832d9
+        securityContext:
+          privileged: true
+        resources:
+          requests:
+            memory: "29Gi"
+
+presubmits:
+  nmstate/kubernetes-nmstate:
+    - <<: *presubmit_job_base
+      name: pull-kubernetes-nmstate-e2e-k8s-1.14.6
+      spec:
+        <<: *presubmit_spec_base
+        containers:
+          - <<: *presubmit_runner_container_base
+            command:
+              - "/usr/local/bin/runner.sh"
+              - "/bin/sh"
+              - "-c"
+              - "automation/check-patch.k8s-1.14.6.sh"
```

Comment:
Could you please make it just `automation/check-patch.k8s.sh`, so we don't need to send PRs every time we bump version. The same for OKD.

Reply:
We need the full path to get the provider, in the future we may want to just parse job name from env https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md#job-environment-variables

Reply:
I don't get why that should be a part of prow config. I would prefer if we had two lanes, OKD and Kubernetes. The change of versions we use would be a mere patch on our repository. No need for renaming of bunch of files under automation/ and opening PRs outside nmstate.

We can replace our parsing of the script path by two scripts, one in the okd.sh, the other in k8s.sh.

Reply:
I am ok with that, but let's make it work with the current automation, then we change it.

Reply:
Or change it on our side now? Should be just a small PR and we should be set for life. Up to you though, I don't mind either option.

Reply:
@phoracek automation is run also at other places... we will have to change that too

Reply:
This configuration is specific for kubernetes-nmstate. Am I missing something? What are "other places"?

Reply:
So steps would be:
- Use check-patch.sh here 
- Configure prow at github knmstate
- Do a PR there that iterate until it passes the prow jobs.

Reply:
You meant use check-patch.k8s.sh? Or check-patch.sh and pass en variable saying it's for kubernetes, but since both require change on knmstate, I'd be in favor of a simple script dedicated for kubernetes

Reply:
just check-patch.sh we are already using if statements at test.sh.

Reply:
And how will we be able to tell which provider should be used?

Reply:
So:
1. you can test jobs outside of prow with `phaino` and `mkpj`: https://github.com/kubevirt/project-infra/blob/31ecc29023fe6acef9821f323b48cf8e521d8678/robots/flakefinder/README.md#run-the-job-locally-using-phaino.
1. You can test jobs in prow by directly posting them (at least gal can do that):
https://github.com/kubevirt/project-infra/tree/master/github/ci#testing-new-prowjobs

Reply:
In both cases you can create a PR in nmstate and point `mkpj` to that PR when you create the job for local or remote run.

Reply:
@gbenhaim would you please change this line to

```
"automation/check-patch.k8s.sh"
```

(@qinqon please stop me if you disagree)

We will do the renaming on our side.

Reply:
nah agree on having two lanes without version.

Reply:
:+1 Fixed